File: frontend/src/lib/task.ts
class Task {

  id: number | undefined;
  title: string;
  content: string;
  create_time: number;
  edit_time: number;
  deadline: undefined | number;
  status: number;
  belong_agenda_id: number;

  constructor(task: {
    id?: number,
    title?: string,
    content?: string,
    create_time?: number,
    edit_time?: number,
    deadline?: undefined | number,
    status: number,
    belong_agenda_id?: number | undefined,
  }) {
    this.id = task.id;
    this.title = task.title ?? '';
    this.content = task.content ?? '';
    this.create_time = task.create_time ?? new Date().getTime();
    this.edit_time = task.edit_time ?? new Date().getTime();
    this.deadline = task.deadline;
    this.status = task.status;
    this.belong_agenda_id = task.belong_agenda_id ?? 0;
  }

}

export default Task;

File: server/entity/src/task.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "tasks")]
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)]
    pub id: i32,
    pub title: String,
    #[sea_orm(column_type = "Text")]
    pub content: String,
    pub create_time: i64,
    pub edit_time: i64,
    // #[serde(skip_serializing_if = "Option::is_none")]
    pub deadline: Option<i64>,
    pub status: i32,
    pub belong_agenda_id: i32,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
